src/tests/functional/plugin/gpu/single_layer_tests/dynamic/reduce.cpp

```diff
@@ -0,0 +1,361 @@
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_l2_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceL2, ReduceLayerGPUTest::getTestCaseName);
+
+
+// ================== Reduction logical types (LogicalOr, LogicalAnd) ==================
+const auto reduceLogicalOr = ::testing::Combine(
+        ::testing::ValuesIn({dyn1d, dyn6d}),
+        ::testing::Values(ElementType::boolean),
+        ::testing::Values(helpers::ReductionType::LogicalOr),
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_logicalor_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceLogicalOr, ReduceLayerGPUTest::getTestCaseName);
+
+const auto reduceLogicalAnd = ::testing::Combine(
+        ::testing::ValuesIn({dyn3d, dyn5d}),
+        ::testing::Values(ElementType::boolean),
+        ::testing::Values(helpers::ReductionType::LogicalAnd),
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_logicaland_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceLogicalAnd, ReduceLayerGPUTest::getTestCaseName);
+
+
+// ================== various reduce-axis ==================
+const std::vector<ReduceInput> dynVariousAxisInputs = {
+    // 4D
+    {
+        {
+            { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}} }
+        },
+        {0}
+    },
+    {
+        {
+            { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}} }
+        },
+        {1, 3}
+    },
+    {
+        {
+            { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}} }
+        },
+        {2, 3}
+    },
+    {
+        {
+            { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}} }
+        },
+        {0, 2, 3}
+    },
+    // 5D
+    {
+        {
+            { {-1, -1, -1, -1, -1}, {{4, 4, 4, 4, 4}, {5, 5, 5, 1, 5}} }
+        },
+        {1}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1}, {{4, 4, 4, 4, 4}, {5, 5, 5, 1, 5}} }
+        },
+        {0, 1}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1}, {{4, 4, 4, 4, 4}, {5, 5, 5, 1, 5}} }
+        },
+        {2, 3, 4}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1}, {{4, 4, 4, 4, 4}, {5, 5, 5, 1, 5}} }
+        },
+        {0, 1, 2, 4}
+    },
+    // 6D
+    {
+        {
+            { {-1, -1, -1, -1, -1, -1}, {{2, 3, 4, 5, 6, 7}, {5, 5, 5, 1, 5, 5}} }
+        },
+        {0}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1, -1}, {{2, 3, 4, 5, 6, 7}, {5, 5, 5, 1, 5, 5}} }
+        },
+        {0, 1}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1, -1}, {{2, 3, 4, 5, 6, 7}, {5, 5, 5, 1, 5, 5}} }
+        },
+        {2, 3, 4, 5}
+    },
+    {
+        {
+            { {-1, -1, -1, -1, -1, -1}, {{2, 3, 4, 5, 6, 7}, {5, 5, 5, 1, 5, 5}} }
+        },
+        {0, 2, 3, 4, 5}
```

Comment:
(random position) please replace one of the axes vector with corresponding vector including minus value